test/chemistry/test_oovqe.py

```diff
@@ -0,0 +1,121 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+""" Test OOVQE """
+
+import unittest
+from test.aqua import QiskitAquaTestCase
+from ddt import ddt
+from qiskit.aqua import aqua_globals
+from qiskit.aqua import QuantumInstance
+from qiskit.aqua.components.optimizers import COBYLA
+from qiskit.chemistry.core import Hamiltonian
+from qiskit.chemistry.drivers import PySCFDriver, UnitsType
+from qiskit.chemistry.components.variational_forms import UCCSD
+from qiskit.chemistry.components.initial_states import HartreeFock
+from qiskit.chemistry.core import TransformationType, QubitMappingType
+from qiskit.chemistry.algorithms.minimum_eigen_solvers import OOVQE
+from qiskit.aqua.operators.expectations import MatrixExpectation
+from qiskit import Aer
+
+
+@ddt
+class TestOOVQE(QiskitAquaTestCase):
+    """ Test of the OOVQE algorithm"""
+
+    def setUp(self):
+        super().setUp()
+        self.energy_vqe = -2.77  # energy of the VQE with pUCCD ansatz and LBFGSB optimizer
+        self.initial_point = [0.039374, -0.47225463, -0.61891996, 0.02598386, 0.79045546,
+                              -0.04134567, 0.04944946, -0.02971617, -0.00374005, 0.77542149]
+        self.seed = 50
+        aqua_globals.random_seed = self.seed
+        driver = PySCFDriver(
+            atom="H 1.738000 .0 .0; H .15148 1.73139 .0; H -1.738 .0 .0; H -0.15148 -1.73139 .0",
+            unit=UnitsType.ANGSTROM,
+            charge=0,
+            spin=0,
+            basis='sto3g')
+        self.qmolecule = driver.run()
+        self.core = Hamiltonian(transformation=TransformationType.FULL,
+                                qubit_mapping=QubitMappingType.PARITY,
+                                two_qubit_reduction=False,
+                                freeze_core=False,
```

Comment:
As there seems to be code specific to dealing with frozen core we should have a test around that and not always use the same configuration over the operator creation.

Reply:
Implemented.